src/collection-view.js

```diff
@@ -641,22 +643,30 @@ const CollectionView = Backbone.View.extend({
     return this;
   },
 
+  addChildViews(views, opts = {}) {
+    if(!_.isArray(view)) return;
+    
+    _.each(views, (view) => this.addChildView(view, opts.index, false));
+    
+    if(this.isRendered() && opts.shouldTriggerShow !== false)
+      this._showChildren();
+  },
+  
   // Render the child's view and add it to the HTML for the collection view at a given index, based on the current sort
-  addChildView(view, index) {
+  addChildView(view, index, shouldTriggerShow = true) {
     if (!view || view._isDestroyed) {
       return view;
     }
 
-    if (!this._isRendered) {
```

Comment:
If we render the children on `addChildView` then `addChildView` is consistently dealing with the children passed to the collection of the `CollectionView` so whether or not the collectionview is "rendered" `addChildView` acts the same and the index is in relation to the `collection` or any child already added.  If the view is not rendered, there is no index or children and `addChildView` acts differently.  This requires quite a bit of understanding in order to know when or how this method can be called.